src/Model/ModelManager.php

```diff
@@ -345,7 +347,7 @@ public function addIdentifiersToQuery(string $class, BaseProxyQueryInterface $qu
         $qb->andWhere(sprintf('( %s )', implode(' OR ', $sqls)));
     }
 
-    public function batchDelete(string $class, BaseProxyQueryInterface $query): void
+    public function batchDelete(string $class, BaseProxyQueryInterface $query, int $batchSize = self::BATCH_SIZE): void
```

Comment:
shouldn't we add this to the interface? or at least with a next major comment?

Reply:
We cant add this to the interface since it wont be bc but yes we can add the next major constant and introduce the constant in the SonataAdminBundle instead

Reply:
I was thinking about the constant in `SonataAdmin`, but I think this must be kept in the scope of each model manager (the behavior regarding the batch size, the way each driver handles the transactions and the cost it demands is specific and not a common concern).

Reply:
If we can't add the constant to the SonataAdmin interface, we can't add the batchSize parameter to the interface too.
I'm fine with both solution.

Reply:
Then what is the use case for adding this parameter on the function? How are you suposed to pass it on an admin?

Reply:
It would be easier to add it to the interface, but even if we don't, people can still pass the batch size as third parameter after a instanceof check.